src/XMakeBuildEngine/Construction/ProjectRootElement.cs

```diff
@@ -1892,7 +1897,11 @@ public void ReloadFrom(string path, bool throwIfUnsavedChanges = true, bool? pre
             ErrorUtilities.VerifyThrowInvalidOperation(File.Exists(path), "FileToReloadFromDoesNotExist", path);
 
             Func<bool, XmlDocumentWithLocation> documentProducer = shouldPreserveFormatting => LoadDocument(path, shouldPreserveFormatting);
-            ReloadFrom(documentProducer, throwIfUnsavedChanges, preserveFormatting);
+
+            // Reloads from a file leave this object up to date wrt to that file, so mark the object as up to date
+            Action postReloadActions = MarkAsUpToDateToDisk;
+
+            ReloadFrom(documentProducer, throwIfUnsavedChanges, preserveFormatting, postReloadActions);
```

Comment:
You're right, looks like needless complexity. I'll remove it.
I was originally going for the "there's one reload method with the core logic and the rest just personalize it" approach, but here it does not look like it's worth it.